### ISSUE TITLE
MD Contact names?

### ISSUE THREAD

**Issue by HUGEIT** [adiwajshing/Baileys#873]
I can only find the push names in the events but what about the names the way I have them saved in my contacts on the phone? in the old connection we had `name` and `notify` but in Baileys multi-device there is only `notify` available?

**Comment by itacirgabral** [comment]
the event `messages.upsert`  type `prepend`  have `messageStubParameters`
the event `contacts.upsert` have `notify`

(I'm not sure about what each one means)

**Comment by HUGEIT** [comment]
just dumped all the data of the events 

writeToFile('contacts.upsert',  JSON.stringify(m))
writeToFile('contacts.update',  JSON.stringify(m))
writeToFile('chats.upsert',  JSON.stringify(m))
writeToFile('groups.update',  JSON.stringify(m))
writeToFile('chats.update',  JSON.stringify(m))
writeToFile('chats.set',  JSON.stringify(m))
writeToFile('messages.upsert',  JSON.stringify(m))
writeToFile('messages.update',  JSON.stringify(m))

and ran a ```grep -R``` on some of my contact names, none showed up except for when their names were written within message contents of course.

am I missing an event or are the phone book names simply not available yet?

**Comment by itacirgabral** [comment]
event **messages.upsert**:
- https://github.com/adiwajshing/Baileys/blob/multi-device/Example/example.ts#L53

type **prepend**:
- https://github.com/adiwajshing/Baileys/blob/multi-device/src/Types/Message.ts#L147

I've had cleaned all my chats before read the qrcode.
Before the automatic close the **prepend** event **messages.upsert** bring me my full agenda's contacts.

----

> am I missing an event or are the phone book names simply not available yet?

I think `messages.upsert |> prepend` just happen once to populate the ui state, read again the QRCode and save it.

**Comment by HUGEIT** [comment]
hey @itacirgabral thank you for responding but can you attach a snippet of how it looks like for you? I may be massively wrong but to me the entries only carry a ```pushName``` property that everyone can set for non-contacts but it does not show the actual name for the phone book.

**Comment by HUGEIT** [comment]
I found out something. _once I change a single name in my phone book_, I get the list of numbers and names.

after changing a name it does ```server_sync critical_unblock_low```, then baileys responds, it goes into processSyncActions etc. and publishes the list.

so I think we have to tell baileys to do that on connect

**Comment by itacirgabral** [comment]
> server_sync critical_unblock_low

I'm dont know

```typescript
socket.ev.on ('messages.upsert', ({ messages, type }) => {
  if (type === 'prepend') {
    const goodOnes = messages.filter(m => m.messageStubType === 99 || m.messageStubType === 65)

    // save it
    console.log(goodOnes, null, 4)
  }
})
```

https://github.com/adiwajshing/Baileys/blob/multi-device/WAProto/index.d.ts#L18650

**Comment by HUGEIT** [comment]
sorry I don't know if we are talking about the same here. I mean the full names, as defined in https://github.com/adiwajshing/Baileys/blob/multi-device/src/Socket/chats.ts#L328

**Comment by itacirgabral** [comment]
```javascript
{
  messageStubParameters: [ 'Magalzão Barreto' ],
  labels: [],
  key: MessageKey {
    remoteJid: '55659318çççç@s.whatsapp.net',
    fromMe: true,
    id: 'FE068E2DC89A7593E67DC3196016B0CF'
  },
  messageTimestamp: Long { low: 1635482519, high: 0, unsigned: true },
  messageStubType: 99
}

{
    messageStubParameters: [ 'Douglas SI' ],
    labels: [],
    key: MessageKey {
      remoteJid: '55658478çççç@s.whatsapp.net',
      fromMe: true,
      id: 'A9770C112434624E4A4E856F13AF081F'
    },
    messageTimestamp: Long { low: 1634100107, high: 0, unsigned: true },
    messageStubType: 65
  
```

`messageStubParameters[0]` returns the name saved on cellphone by the user.

**Comment by HUGEIT** [comment]
ok that really does look like what I am after, you are correct. I have no idea why I am not getting those but I have finally figured out why Baileys does not receive the list that Baileys itself expects on first connection.

it is an issue I have commented on before but for another reason, #834 

@pokearaujo is correct - at the time of the connection there is something missing, this is why first try won't get us keys and instead throws.

I will close this issue here with reference to #834
